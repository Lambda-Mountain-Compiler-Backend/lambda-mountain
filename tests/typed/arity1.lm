
f := λx. x;
f := λx y. y;
f := λx y z. z;

main := (print-s (f 1 2));
