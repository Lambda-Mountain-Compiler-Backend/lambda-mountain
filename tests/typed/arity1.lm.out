((((() (Global (f (: ((App ((Variable x) (: ((Variable x) S)))) (-> (S S))))))) (Global (f (: ((App ((App ((Variable x) (Variable y))) (: ((Variable y) S)))) (-> ((Cons (S S)) S))))))) (Global (f (: ((App ((App ((App ((Variable x) (Variable y))) (Variable z))) (: ((Variable z) S)))) (-> ((Cons ((Cons (S S)) S)) S))))))) (GExpr (: ((App ((: ((Variable f) (And ((And ((-> (S S)) (-> ((Cons (S S)) S)))) (-> ((Cons ((Cons (S S)) S)) S)))))) (: ((App ((: ((Literal 1) Atom)) (: ((Literal 2) Atom)))) (Cons (Atom Atom)))))) S))))